Sources/Sentry/SentryMetricKitIntegration.m

```diff
@@ -282,9 +282,14 @@ - (void)buildAndCaptureMXEventFor:(NSArray<SentryMXFrame *> *)rootFrames
             if (noChildren && lastUnprocessedSibling) {
                 [self captureEventNotPerThread:stackTraceFrames params:params];
 
-                // Pop all siblings
-                for (int i = 0; i < parentFrame.subFrames.count; i++) {
+                if (parentFrame == nil) {
+                    // No parent frames
                     [stackTraceFrames removeLastObject];
+                } else {
+                    // Pop all siblings
+                    for (int i = 0; i < parentFrame.subFrames.count; i++) {
+                        [stackTraceFrames removeLastObject];
+                    }
```

Comment:
I don't quite understand what the objective is here, after reading the diagram in the headerdoc comment for the method (thanks to whomever created that btw!), i'm not sure why this example gets a result that preserved frames 4 and 5 and not 2:
```
 * | frame 0 |
 *      | frame 1 |
 *          | frame 2 |
 *          | frame 3 |
 *              | frame 4 |
 *              | frame 5 |
 *              | frame 6 |     -> stack trace consists of [0, 1, 3, 4, 5, 6]
 ```
---
Is it describing a the following call graph (a):
```
time ----->

|---frame 0------------------------------------------|
  |--frame 1-----------------------------------------|
    |-frame 2-| |-frame 3----------------------------|
                  |-frame 4-| |-frame 5-| |-frame 6-|
```
which would  be a tree like
```
     0
     |
     1
    / \
   2   3
      /|\
     4 5 6
```
---
or is it like this one (b)
```
time ----->

|---frame 0------------------------------------------|
  |--frame 1-----------------------------------------|
    |-frame 2-| |-frame 3----------------------------|
                  |-frame 4-----| 
                   |-frame 5---| 
                    |-frame 6-|
```
which would have a tree like
```
     0
     |
     1
    / \
   2   3
       |
       4
       |
       5
       |
       6
```
because the way the headerdoc diagram describes it sounds like (a) to me, but the result makes me think it's like (b). 
---
So I guess I'm just looking for a little more clarification on what "pop siblings" means, because I expected it to either remove 4 and 5 or preserve 2. It might be helpful to also include some diagrams describing what each test case is doing for an existing test and the new one you wrote.

Reply:
I created the diagram. It's b. They aren't siblings in terms of the tree but in terms of the stack trace level nesting. I'm going to update the code docs. Thanks for pointing that out so clearly and easy to understand @armcknight.

Reply:
I know it's a bit weird, but that's what my observations with MetricKit data revealed. I initially wrote the logic as you pointed out with A, but that didn't work properly. A also seemed more logical to me.